src/Mgate/PubliBundle/Controller/DocumentController.php

```diff
@@ -65,24 +66,18 @@ public function voirAction(Document $documentType)
 
     /**
      * @Security("has_role('ROLE_SUIVEUR')")
+     * @param Request $request
+     * @param Etude $etude
+     * @return bool|\Symfony\Component\HttpFoundation\RedirectResponse|\Symfony\Component\HttpFoundation\Response
      */
-    public function uploadEtudeAction(Request $request, $etude_id)
+    public function uploadEtudeAction(Request $request, Etude $etude)
     {
-        $em = $this->getDoctrine()->getManager();
-        $etude = $em->getRepository('MgateSuiviBundle:Etude')->getByNom($etude_id);
-
-        if (!$etude) {
-            throw $this->createNotFoundException('Le document ne peut être lié à une étude qui n\'existe pas!');
-        }
-
         if ($this->get('Mgate.etude_manager')->confidentielRefus($etude, $this->getUser(), $this->get('security.authorization_checker'))) {
             throw new AccessDeniedException('Cette étude est confidentielle !');
         }
 
-        $options['etude'] = $etude;
-
-        if (!$response = $this->upload($request, false, $options)) {
-            // Si tout est ok
+        if (!$response = $this->upload($request, false, ['etude' => $etude])) {
+            $this->addFlash('success','Document mis en ligne');
             return $this->redirect($this->generateUrl('MgateSuivi_etude_voir', array('nom' => $etude->getNom())));
```

Comment:
Yes. However, there are plenty of `array()` calls. 
It is a non sense to fix only these ones. I will apply php-cs-fixer in a few days to fix all that.